src/components/Search/SearchRouter/SearchRouterList.tsx

```diff
@@ -0,0 +1,149 @@
+        return true;
+    }
+    return false;
+}
+
+function SearchRouterItem(props: UserListItemProps<OptionData> | SearchQueryListItemProps) {
+    const styles = useThemeStyles();
+
+    if (isSearchQueryListItem(props)) {
+        return (
+            <SearchQueryListItem
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...props}
+            />
+        );
+    }
+    return (
+        <UserListItem
+            pressableStyle={styles.br2}
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...props}
+        />
+    );
+}
+
+function SearchRouterList(
+    {currentQuery, reportForContextualSearch, recentSearches, recentReports, onSearchSubmit, updateUserSearchQuery, closeAndClearRouter}: SearchRouterListProps,
+    ref: ForwardedRef<SelectionListHandle>,
+) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const sections: Array<SectionListDataType<OptionData | SearchQueryItem>> = [];
+
+    if (currentQuery?.inputQuery) {
+        sections.push({
+            data: [
+                {
+                    text: currentQuery?.inputQuery,
+                    singleIcon: Expensicons.MagnifyingGlass,
+                    query: currentQuery?.inputQuery,
+                    itemStyle: styles.activeComponentBG,
+                    keyForList: 'findItem',
+                },
+            ],
+        });
+    }
+
+    if (reportForContextualSearch) {
+        sections.push({
+            data: [
+                {
+                    text: `${translate('search.searchIn')}${reportForContextualSearch.text ?? reportForContextualSearch.alternateText}`,
+                    singleIcon: Expensicons.MagnifyingGlass,
+                    query: `in:${reportForContextualSearch.reportID}`,
+                    itemStyle: styles.activeComponentBG,
+                    keyForList: 'contextualSearch',
+                    isContextualSearchItem: true,
+                },
+            ],
+        });
+    }
+
+    const recentSearchesData = recentSearches?.map(({query}) => ({
+        text: query,
+        singleIcon: Expensicons.History,
+        query,
+        keyForList: query,
+    }));
+
+    if (recentSearchesData && recentSearchesData.length > 0) {
+        sections.push({title: translate('search.recentSearches'), data: recentSearchesData});
+    }
+
+    const styledRecentReports = recentReports.map((item) => ({...item, pressableStyle: styles.br2}));
+    sections.push({title: translate('search.recentChats'), data: styledRecentReports});
+
+    const onSelectRow = useCallback(
+        (item: OptionData | SearchQueryItem) => {
+            if (!('query' in item) || !item.query) {
+                // Handle selection of "Recent chat"
+                closeAndClearRouter();
+                if ('reportID' in item && item?.reportID) {
```

Comment:
Maybe refactoring this conditionals into a "type guard" might make our intentions clearer, e.g. `isReportItem` or `isLoginItem`